### PR TITLE
Marylin/whack a mole

### PR DESCRIPTION
**Pull request by marylin-moya**

### CODE REVIEW

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        System.out.println("----------------------------");
+        whackAMole.printGrid();
+
+        System.out.println("-----You have a maximum of " + numAttemps + " attempts to get all the moles wacked-----");
+        int xAxis;
+        int yAxis;
+        Scanner xScanner;
+        Scanner yScanner;
+        for (int i = numAttemps; i > 0; i++) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: BLOCKER Correct this loop's end condition. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2189)
MAJOR "i" is incremented and will never reach "stop condition". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2251)
MINOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            xScanner = new Scanner(System.in);
+            System.out.print("Enter the x coordinates: ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    void printGridToUser() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                if (this.moleGrid[i][j] == 'M') {
+                    System.out.print("*  ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        int yAxis;
+        Scanner xScanner;
+        Scanner yScanner;
+        for (int i = numAttemps; i > 0; i++) {
+            xScanner = new Scanner(System.in);
+            System.out.print("Enter the x coordinates: ");
+            xAxis = xScanner.nextInt();
+
+            yScanner = new Scanner(System.in);
+            System.out.print("Enter the y coordinates: ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    void printGridToUser() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                if (this.moleGrid[i][j] == 'M') {
+                    System.out.print("*  ");
+                } else {
+                    System.out.print(this.moleGrid[i][j] + "  ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                }
+            }
+            System.out.println("");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            xScanner = new Scanner(System.in);
+            System.out.print("Enter the x coordinates: ");
+            xAxis = xScanner.nextInt();
+
+            yScanner = new Scanner(System.in);
+            System.out.print("Enter the y coordinates: ");
+            yAxis = yScanner.nextInt();
+
+            if (xAxis == -1 || yAxis == -1) {
+                System.out.println("Giving up!!");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        //create a 10 by 10 grid
+        WhackAMole whackAMole = new WhackAMole(numAttemps, gridLenght);
+
+        //Place randomly a total of 10 moles
+        for (int i = 0; i < numberOfMoles; i++) {
+            Random rand = new Random();
+            int x = rand.nextInt(numberOfMoles);
+            int y = rand.nextInt(numberOfMoles);
+            if (!whackAMole.place(x, y)) {
+                System.out.println("There is already a mole at " + x + "," + y + ". Trying again!");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+                i--;
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Refactor the code in order to not assign to this loop counter from within the loop body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+            yScanner = new Scanner(System.in);
+            System.out.print("Enter the y coordinates: ");
+            yAxis = yScanner.nextInt();
+
+            if (xAxis == -1 || yAxis == -1) {
+                System.out.println("Giving up!!");
+                whackAMole.printGrid();
+                break;
+            } else {
+                System.out.println("State of Game");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+        //Place randomly a total of 10 moles
+        for (int i = 0; i < numberOfMoles; i++) {
+            Random rand = new Random();
+            int x = rand.nextInt(numberOfMoles);
+            int y = rand.nextInt(numberOfMoles);
+            if (!whackAMole.place(x, y)) {
+                System.out.println("There is already a mole at " + x + "," + y + ". Trying again!");
+                i--;
+            } else {
+                System.out.println("Mole places at " + x + "," + y);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+        }
+    }
+
+    /**
+     * Method to print the grid showing the status of the game even moles.
+     */
+    void printGrid() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                System.out.print(this.moleGrid[i][j] + "  ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+                whackAMole.printGrid();
+                break;
+            } else {
+                System.out.println("State of Game");
+                whackAMole.whack(xAxis, yAxis);
+                whackAMole.printGridToUser();
+            }
+
+            if (whackAMole.molesLeft == 0) {
+                System.out.println("All moles were whacked with score: " + whackAMole.score);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+
+    /**
+     * Method to print the grid showing the status of the game even moles.
+     */
+    void printGrid() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                System.out.print(this.moleGrid[i][j] + "  ");
+            }
+            System.out.println("");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+            } else {
+                System.out.println("Mole places at " + x + "," + y);
+            }
+        }
+
+        //Try to whack 50 times the moles given the x and y
+        //Finish the game if we enter xAxis -1 and yAxis -1 printing the grid
+        //The game ends if the user is able to uncover all the moles or if the user runs out of attempts.
+
+        System.out.println("----------------------------");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+                whackAMole.printGridToUser();
+            }
+
+            if (whackAMole.molesLeft == 0) {
+                System.out.println("All moles were whacked with score: " + whackAMole.score);
+                break;
+            }
+
+            if (whackAMole.attemptsLeft == 0) {
+                System.out.println("There are no more attemps with score: " + whackAMole.score);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        //Try to whack 50 times the moles given the x and y
+        //Finish the game if we enter xAxis -1 and yAxis -1 printing the grid
+        //The game ends if the user is able to uncover all the moles or if the user runs out of attempts.
+
+        System.out.println("----------------------------");
+        whackAMole.printGrid();
+
+        System.out.println("-----You have a maximum of " + numAttemps + " attempts to get all the moles wacked-----");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

**Comment by carledriss** [comment]
SonarQube analysis reported 19 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 16 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+        int xAxis;
+        int yAxis;
+        Scanner xScanner;
+        Scanner yScanner;
+        for (int i = numAttemps; i > 0; i--) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+            xScanner = new Scanner(System.in);
+            System.out.print("Enter the x coordinates: ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem:rowGrid) {
+                if (charItem == 'M') {
+                    strBuilderRow.append("*  ");
+                } else {
+                    strBuilderRow.append(charItem);
+                    strBuilderRow.append("  ");
+                }
+            }
+            LOGGER.info(strBuilderRow.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+     * Method to print the grid showing the status of the game even moles.
+     */
+    private void printGrid() {
+        for (char[] rowGrid:this.moleGrid) {
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem:rowGrid) {
+                strBuilderRow.append(charItem);
+                strBuilderRow.append("  ");
+            }
+            LOGGER.info(strBuilderRow.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by carledriss** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 11 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 22]
```diff
@@ -0,0 +1,174 @@
...
+                strBuilderMessage.append(y);
+                strBuilderMessage.append(". Trying again!");
+                i--;
+            } else {
+                strBuilderMessage.append("Mole places at ");
+                strBuilderMessage.append(x);
+                strBuilderMessage.append(",");
+                strBuilderMessage.append(y);
+            }
+            LOGGER.info(strBuilderMessage.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 23]
```diff
@@ -0,0 +1,174 @@
...
+                whackAMole.printGrid();
+                break;
+            } else {
+                LOGGER.info("State of Game");
+                whackAMole.whack(xAxis, yAxis);
+                whackAMole.printGridToUser();
+            }
+
+            if (whackAMole.molesLeft == 0) {
+                LOGGER.info("All moles were whacked with score: " + whackAMole.score);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 24]
```diff
@@ -0,0 +1,174 @@
...
+                break;
+            }
+
+            if (whackAMole.attemptsLeft == 0) {
+                LOGGER.info("There are no more attemps with score: " + whackAMole.score);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 25]
```diff
@@ -0,0 +1,174 @@
...
+            int x = rand.nextInt(numberOfMoles);
+            int y = rand.nextInt(numberOfMoles);
+            StringBuilder strBuilderMessage = new StringBuilder();
+            if (!whackAMole.place(x, y)) {
+                strBuilderMessage.append("There is already a mole at ");
+                strBuilderMessage.append(x);
+                strBuilderMessage.append(",");
+                strBuilderMessage.append(y);
+                strBuilderMessage.append(". Trying again!");
+                i--;
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Refactor the code in order to not assign to this loop counter from within the loop body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 26]
```diff
@@ -0,0 +1,174 @@
...
+            LOGGER.info(strBuilderMessage.toString());
+        }
+
+        //Try to whack 50 times the moles given the x and y
+        //Finish the game if we enter xAxis -1 and yAxis -1 printing the grid
+        //The game ends if the user is able to uncover all the moles or if the user runs out of attempts.
+
+        LOGGER.info("-------------------------------------");
+        whackAMole.printGrid();
+        LOGGER.info("-----You have a maximum of " + numAttemps + " attempts to get all the moles wacked-----");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 27]
```diff
@@ -0,0 +1,174 @@
...
+     * Method to print the grid showing the status of the game even moles.
+     */
+    private void printGrid() {
+        for (char[] rowGrid : this.moleGrid) {
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem : rowGrid) {
+                strBuilderRow.append(charItem);
+                strBuilderRow.append("  ");
+            }
+            LOGGER.info(strBuilderRow.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by carledriss** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 28]
```diff
@@ -0,0 +1,175 @@
...
+                strBuilderMessage.append(y);
+                strBuilderMessage.append(". Trying again!");
+                i--;
+            } else {
+                strBuilderMessage.append("Mole places at ");
+                strBuilderMessage.append(x);
+                strBuilderMessage.append(",");
+                strBuilderMessage.append(y);
+            }
+            LOGGER.fine(strBuilderMessage.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 29]
```diff
@@ -0,0 +1,175 @@
...
+     * Method to print the grid showing the status of the game even moles.
+     */
+    private void printGrid() {
+        for (char[] rowGrid : this.moleGrid) {
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem : rowGrid) {
+                strBuilderRow.append(charItem);
+                strBuilderRow.append("  ");
+            }
+            LOGGER.fine(strBuilderRow.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 30]
```diff
@@ -0,0 +1,175 @@
...
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem : rowGrid) {
+                if (charItem == 'M') {
+                    strBuilderRow.append("*  ");
+                } else {
+                    strBuilderRow.append(charItem);
+                    strBuilderRow.append("  ");
+                }
+            }
+            LOGGER.fine(strBuilderRow.toString());
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by carledriss** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 31]
```diff
@@ -0,0 +1,175 @@
...
+            this.score++;
+            this.molesLeft--;
+        }
+        this.attemptsLeft--;
+    }
+
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    private void printGridToUser() {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: In order to test printGridToUser method
You should return a String

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 32]
```diff
@@ -0,0 +1,175 @@
...
+            }
+            LOGGER.fine(strBuilderRow.toString());
+        }
+    }
+
+
+    /**
+     * Method to print the grid showing the status of the game even moles.
+     */
+    private void printGrid() {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: In order to test printGridToUser method
You should return a String

File: src/main/java/org/minions/devfund/marylin/WhackAMole.java [hunk 33]
```diff
@@ -0,0 +1,175 @@
...
+            LOGGER.fine(strBuilderRow.toString());
+        }
+    }
+
+    /**
+     * Main method to play the game.
+     *
+     * @param args Default Argument.
+     */
+    public static void main(final String[] args) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/marylin/WhackAMole.java]: Since edx doesn't require this method.
You could remove it!

**Comment by marylin-moya** [comment]
week1_progress
week1_wackamolle_report

**Review by carledriss — CHANGES_REQUESTED** [review]
Fix findbugs issues.

**Review by carledriss — APPROVED** [review]
LGTM! 👍

**Pull request merged**